backend/src/packages/orders/order.repository.ts

```diff
@@ -24,72 +27,111 @@ class OrderRepository implements Omit<IRepository, 'find'> {
-      where: (orders) => eq(orders.id, id),
-    });
-
-    return (result ?? null) && OrderEntity.initialize(result as OrderEntityT);
+        truck: {
+          id: schema.shifts.truckId,
+          licensePlateNumber: schema.drivers.driverLicenseNumber,
+        },
+      })
+      .from(this.ordersSchema)
+      .innerJoin(schema.shifts, eq(this.ordersSchema.shiftId, schema.shifts.id))
+      .innerJoin(schema.users, eq(schema.shifts.driverId, schema.users.id))
+      .innerJoin(
+        schema.drivers,
+        eq(schema.drivers.userId, schema.shifts.driverId),
+      )
+      .where(eq(this.ordersSchema.id, id));
+
+    return order;
   }
 
-  public async find(
-    search: Partial<Pick<OrderEntityT, 'userId' | 'driverId' | 'businessId'>>,
-  ): Promise<OrderEntity[]> {
-    const orders = await this.db
+  public async findAllBusinessOrders(
+    search: Partial<{
+      userId: OrderEntityT['userId'];
+      driverId: UserEntityT['id'];
+      businessId: OrderEntityT['businessId'];
+    }>,
+  ): Promise<OrderEntityT[]> {
+    return await this.db
       .driver()
-      .select()
+      .select({
+        id: this.ordersSchema.id,
+        userId: this.ordersSchema.userId,
+        businessId: this.ordersSchema.businessId,
+        price: this.ordersSchema.price,
+        scheduledTime: this.ordersSchema.scheduledTime,
+        startPoint: this.ordersSchema.startPoint,
+        endPoint: this.ordersSchema.endPoint,
+        status: this.ordersSchema.status,
+        carsQty: this.ordersSchema.carsQty,
+        customerName: this.ordersSchema.customerName,
+        customerPhone: this.ordersSchema.customerPhone,
+        shift: {
+          id: this.ordersSchema.shiftId,
+        },
+        driver: {
+          id: schema.shifts.driverId,
+          firstName: schema.users.firstName,
+          lastName: schema.users.lastName,
+          email: schema.users.email,
+          phone: schema.users.phone,
+          driverLicenseNumber: schema.drivers.driverLicenseNumber,
+        },
+        truck: {
+          id: schema.shifts.truckId,
+          licensePlateNumber: schema.drivers.driverLicenseNumber,
+        },
+      })
       .from(this.ordersSchema)
+      .innerJoin(schema.shifts, eq(this.ordersSchema.shiftId, schema.shifts.id))
+      .innerJoin(schema.users, eq(schema.shifts.driverId, schema.users.id))
+      .innerJoin(
+        schema.drivers,
+        eq(schema.drivers.userId, schema.shifts.driverId),
+      )
       .where(
         combineFilters<DatabaseSchema['orders']>(this.ordersSchema, search),
       );
-
-    return orders.map((it) => OrderEntity.initialize(it as OrderEntityT));
   }
 
-  public async create(entity: OrderEntity): Promise<OrderEntity> {
-    const [result] = await this.db
+  public async create(
+    entity: Omit<OrderEntityT, 'id' | 'shift' | 'driver' | 'truck'> & {
+      shiftId: number;
+    },
+  ): Promise<OrderDatabaseModel | null> {
+    const [result = null] = await this.db
```

Comment:
It can be either error, if some DB constraint is violated, or created DB record. In case of order schema, there're no constraints, so there can't be such an error. Hence, it is guaranteed to return inserted record. Also, other repositories' create method do not have an option of returning null as a result.
```suggestion
  ): Promise<OrderDatabaseModel> {
    const [result] = await this.db
```